src/QoiSharp/QoiImage.cs

```diff
@@ -10,8 +10,12 @@ public class QoiImage
     /// <summary>
     /// Raw pixel data.
     /// </summary>
-    public byte[] Data { get; }
-    
+    public ReadOnlyMemory<byte> Data { get; }
+
+    // Make the zero-copy version available internally for safety. The public API clones the byte[] to guarantee immutability.
+    internal static QoiImage FromMemory(ReadOnlyMemory<byte> memory, int width, int height, Channels channels, ColorSpace colorSpace = ColorSpace.SRgb)
+        => new QoiImage(memory, width, height, channels, colorSpace);
```

Comment:
This could be made public, though some docs would be needed to ensure that users of the API are aware that the memory is shared, not duplicated. If someone passes in a byte[] (which implicitly casts to ReadOnlyMemory<byte>) and then mutates the buffer, they'll corrupt the image.